src/vis/plots/forecast_map.py

```diff
@@ -0,0 +1,361 @@
+                "Incorrect MacroSaude geocode(s). Example: [1101, 1102]"
+            )
+
+    if uf:
+        try:
+            macros_saude = GeoMacroSaude.objects.filter(
+                state=State.objects.get(uf=uf)
+            )
+        except State.DoesNotExist:
+            raise ValueError("Incorrect UF. Example: 'SP'")
+
+    if macroregion:
+        macros_saude = GeoMacroSaude.objects.filter(
+            state__in=(
+                State.objects.filter(
+                    macroregion=(
+                        Macroregion.objects.get(geocode=str(macroregion))
+                    )
+                )
+            )
+        )
+        if not macros_saude:
+            raise ValueError("Incorrect macroregion selected. Options: 1 to 5")
+
+    results = set()
+    for macro in macros_saude:
+        results.add(
+            get_object_or_404(ResultsProbForecast, date=date, geocode=macro)
+        )
+
+    df = pd.concat([obj_to_dataframe(o) for o in results]).reset_index(
+        drop=True
+    )
+
+    df_macro = gpd.GeoDataFrame(
+        pd.concat(
+            [geo_obj_to_dataframe(o) for o in macros_saude],
+        )
+        .drop_duplicates()
+        .reset_index(drop=True)
+    )
+
+    df["date"] = df["date"].apply(str)
+
+    df.prob_low = -df.prob_low
+
+    df["prob_color"] = df.apply(
+        lambda x: x.prob_low
+        if abs(x.prob_low) > abs(x.prob_high)
+        else x.prob_high,
+        axis=1,
+    )
+
+    df["prob_color"] = df.prob_color.apply(lambda x: 0 if abs(x) < 50 else x)
+
+    df = df.drop(columns="id")
+
+    df_macro["state_code"] = df_macro["state"]
+
+    df_macro["state"] = df_macro["state"].astype(int).replace(code_to_state)
+
+    df_macro = df_macro.merge(df, on="geocode", how="left")
+
+    df_macro = gpd.GeoDataFrame(df_macro)
+
+    df_macro["desc_prob"] = np.nan
+
+    df_macro.loc[df_macro.prob_color > 0, "desc_prob"] = str(
+        _("Probabilidade de a incidência superar o limiar histórico")
+    )
+
+    df_macro.loc[df_macro.prob_color < 0, "desc_prob"] = str(
+        _(
+            "Probabilidade de a incidência ser abaixo do limiar inferior histórico"
+        )
+    )
+
+    df_macro.loc[df_macro.prob_color == 0, "desc_prob"] = "N/A"
+
+    maps = macro_maps(
+        df_macro,
+        request=request,
+        title_left=title_left,
+        title_right=title_right,
+        width=width,
+        height=height,
+        fontsize=fontsize,
+    )
+
+    table = macro_table(
+        df_macro,
+        width=table_width,
+        fontsize=fontsize,
+    )
+
+    final_plot = alt.vconcat(maps, table).configure_view(stroke=None)
+
+    return final_plot
+
+
+def macro_maps(
+    df: gpd.GeoDataFrame,
+    request: HttpRequest,
+    title_left: str = "Limiar superior de Incidência na semana",
+    title_right: str = "Previsão probabilística na semana",
+    width: int = 450,
+    height: int = 350,
+    fontsize: int = 16,
+):
+    watermark = (
```

Comment:
Since the watermark is used in various charts, we should move lines 158-186 into a separate function that would just generate the watermark for layering with other charts.